shortestBridge.js

```diff
@@ -0,0 +1,187 @@
+/*
+We're going to approach this problem with a multi-source BFS. The grid will contain
+two islands, so we can first gather all the locations of the first island only.
+
+Once we have the locations of the first island, we can perform our BFS by adding 
+them to the queue with a starting path of 0. Since we're trying to find the shortest path 
+that connects with second island, starting with a path of 0 for all locations will be ideal 
+because with a BFS, you will always have the shorest path if you're visiting a 
+location for the first time.
+
+Once we reach a part of the second island, we will return the current path.
+We will keep track of off all visited locations with a visited set and only update
+our path if it's the very first time visiting that location.
+*/
+
+/*
+Question:
+You are given an n x n binary matrix grid where 1 represents land and 0 represents water.
+
+An island is a 4-directionally connected group of 1's not connected to any other 1's. There are exactly two islands in grid.
+
+You may change 0's to 1's to connect the two islands to form one island.
+
+Return the smallest number of 0's you must flip to connect the two islands.
+*/
+
+const { Queue } = require('../../../utils/queue');
+
+const LAND = 1;
+
+const directions = [
+    [0, 1],
+    [1, 0],
+    [0, -1],
+    [-1, 0],
+];
+
+// 0 1 0
+// 0 0 0
+// 0 0 1
+
```

Comment:
P: Suppose we modified the question to say that you are allowed to move diagonally in all 4 directions in addition to up, down, left, right. How would you change your code to support this?

Reply:
I believe it would look something like this:

`const directions = [
    [0, 1],
    [1, 0],
    [0, -1],
    [-1, 0],
    [1, 1],
    [1, -1],
    [-1, -1],
    [-1, 1],
];`

Reply:
Correct.